tools/devops/automation/templates/configure-build.yml

```diff
@@ -0,0 +1,47 @@
+# This job will parse all the labels present in a PR, will set
+# the tags for the build AND will set a number of configuration
+# variables to be used by the rest of the projects
+steps:
+- checkout: self          # https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#checkout
+  clean: true             # Executes: git clean -ffdx && git reset --hard HEAD
+  submodules: recursive
+
+- pwsh: |
+    Import-Module $Env:SYSTEM_DEFAULTWORKINGDIRECTORY/tools/devops/automation/scripts/GitHub.psm1
+
+    $buildReason = "$(Build.Reason)"
+    $buildSourceBranchName = "$(Build.SourceBranchName)"
+    $buildSourceBranch = "$(Build.SourceBranch)"
+
+    Write-Host "buildReason: ${buildReason}"
+    Write-Host "buildSourceBranchName: ${buildSourceBranchName}"
+    Write-Host "buildSourceBranch: $buildSourceBranch"
+
+    if ($buildReason -eq 'PullRequest') {
+      Write-Host "Configuring build from PR."
+      # This is an interesting step, we do know we are dealing with a PR, but we need the PR id to
+      # be able to get the labels, the buildSourceBranch follows the pattern: refs/pull/{ChangeId}/merge
+      # we could use a regexp but then we would have two problems instead of one
+      $changeId = $buildSourceBranch.Replace("refs/pull/", "").Replace("/merge", "")
```

Comment:
There's actually a variable that will give you this commit id without having to parse it out:  $(System.PullRequest.SourceCommitId)

Reply:
awesome.

Reply:
Also note, if you need the source branch, in the context of a PR trigger, you can also use `$(System.PullRequest.SourceBranch)`, which i believe gives the short name of the topic branch and not the `merge` branch

Reply:
If `$(Build.Reason)` is `manual` (as with all other trigger types in the `else` case) then use `$(Build.SourceVersion)`